src/vsr/replica_test.zig

```diff
@@ -833,6 +836,74 @@ test "Cluster: view-change: duel of the primaries" {
     try expectEqual(t.replica(.R2).commit(), 3);
 }
 
+test "Cluster: view_change: lagging replica advances checkpoint during view change" {
```

Comment:
Consider writing a short paragraph at the begining of the test to explain the high-level picture for _why_ we need this behavior. See, eg, `"Cluster: sync: slightly lagging replica"`.

Here, I think it would be something like this:

> It could be the case that the replica with the most advanced checkpoint has its checkpoint corrupted. In this case, a replica with a slightly older checkpoint must step up as a primary